odl/operator/operator.py

```diff
@@ -334,19 +563,21 @@ def __call__(self, x, out=None, *args, **kwargs):
                                 ''.format(out, self.range, self))
 
             if self.is_functional:
-                raise TypeError('`out` parameter cannot be used'
+                raise TypeError('`out` parameter cannot be used '
                                 'when range is a field')
 
-            self._apply(x, out, *args, **kwargs)
+            self._call_in_place(x, out=out, **kwargs)
             return out
 
         else:  # Out-of-place evaluation
-            result = self._call(x, *args, **kwargs)
+            result = self._call_out_of_place(x, **kwargs)
 
             if result not in self.range:
-                raise TypeError('result {!r} not an element of the range {!r} '
-                                'of {!r}.'
-                                ''.format(result, self.range, self))
+                try:
+                    result = self.range.element(result)
+                except (TypeError, ValueError):
```

Comment:
We can `from future.utils import raise_from` and use that to implement the exception chaining. Good idea.